File: API/Models/Entities/Report.cs
using Models.Base;

namespace Models.Entities
{
    public class Report : BaseEntity
    {
        public int NewMusewaveTrackCount { get; set; }
        public int NewJamendoTrackCount { get; set; }
        public int NewUserCount { get; set; }
        public int NewArtistCount { get; set; }
        public int DailyLoginCount { get; set; }
        public int MonthlyJamendoApiActivity { get; set; }
        public int MonthlyTimeListened { get; set; }
        public decimal MonthlyDonationsAmount { get; set; }
        public int MonthlyDonationsCount { get; set; }
        public int ReportMonth { get; set; }
        public int ReportYear { get; set; }
        public DateTime ReportDate { get; set; }
    }
}

File: Frontend/admin_app/lib/models/report.dart
import 'package:admin_app/models/base/base_entity.dart';
import 'package:json_annotation/json_annotation.dart';

part 'report.g.dart';

@JsonSerializable()
class Report extends BaseEntity {
  int newMusewaveTrackCount;
  int newJamendoTrackCount;
  int newUserCount;
  int newArtistCount;
  int dailyLoginCount;
  int monthlyJamendoApiActivity;
  int monthlyTimeListened;
  double monthlyDonationsAmount;
  int monthlyDonationsCount;
  int reportMonth;
  int reportYear;
  DateTime reportDate;

  Report({
    required super.id,
    required super.createdAt,
    required super.updatedAt,
    required this.newMusewaveTrackCount,
    required this.newJamendoTrackCount,
    required this.newUserCount,
    required this.newArtistCount,
    required this.dailyLoginCount,
    required this.monthlyJamendoApiActivity,
    required this.monthlyTimeListened,
    required this.monthlyDonationsAmount,
    required this.monthlyDonationsCount,
    required this.reportMonth,
    required this.reportYear,
    required this.reportDate,
  });

  factory Report.fromJson(Map<String, dynamic> json) => _$ReportFromJson(json);
  Map<String, dynamic> toJson() => _$ReportToJson(this);
}
